crates/iceberg-catalog/src/service/endpoint_hooks.rs

```diff
@@ -0,0 +1,354 @@
+    ) {
+        futures::future::join_all(self.0.iter().map(|hook| {
+            hook.rename_view(
+                warehouse_id,
+                view_id,
+                request.clone(),
+                request_metadata.clone(),
+            )
+        }))
+        .await;
+    }
+
+    async fn rename_table(
+        &self,
+        warehouse_id: WarehouseId,
+        table_id: TableId,
+        request: Arc<RenameTableRequest>,
+        request_metadata: Arc<RequestMetadata>,
+    ) {
+        futures::future::join_all(self.0.iter().map(|hook| {
+            hook.rename_table(
+                warehouse_id,
+                table_id,
+                request.clone(),
+                request_metadata.clone(),
+            )
+        }))
+        .await;
+    }
+
+    async fn undrop_tabular(
+        &self,
+        warehouse_id: WarehouseId,
+        request: Arc<UndropTabularsRequest>,
+        responses: Arc<Vec<UndropTabularResponse>>,
+        request_metadata: Arc<RequestMetadata>,
+    ) {
+        futures::future::join_all(self.0.iter().map(|hook| {
+            hook.undrop_tabular(
+                warehouse_id,
+                request.clone(),
+                responses.clone(),
+                request_metadata.clone(),
+            )
+        }))
+        .await;
+    }
+}
+
+/// `EndpointHooks` is a trait that allows for custom hooks to be executed within the context of
+/// various endpoints.
+///
+/// The default implementation of every hook does nothing. Override any function if you want to
+/// implement it.
+///
+/// An implementation should be light-weight, ideally every longer running task is deferred to a
+/// background task via a channel or is spawned as a tokio task.
+///
+/// The `EndpointHooks` are passed into the services via the `EndpointHookCollection`. If you want
+/// to provide your own implementation, you'll have to fork and modify the main function to include
+/// your hooks.
+#[async_trait::async_trait]
+pub trait EndpointHooks: Send + Sync + Debug + Display {
+    async fn commit_transaction(
+        &self,
+        _warehouse_id: WarehouseId,
+        _request: Arc<CommitTransactionRequest>,
+        _responses: Arc<Vec<CommitContext>>,
+        _table_ident_map: Arc<HashMap<TableIdent, TableId>>,
+        _request_metadata: Arc<RequestMetadata>,
+    ) {
+        // Default implementation does nothing
+    }
+
+    async fn drop_table(
+        &self,
+        _warehouse_id: WarehouseId,
+        _parameters: TableParameters,
+        _drop_params: DropParams,
+        _table_id: TableId,
+        _request_metadata: Arc<RequestMetadata>,
+    ) {
+    }
+    async fn register_table(
+        &self,
+        _warehouse_id: WarehouseId,
+        _parameters: NamespaceParameters,
+        _request: Arc<RegisterTableRequest>,
+        _metadata: Arc<TableMetadata>,
+        _request_metadata: Arc<RequestMetadata>,
+    ) {
+    }
+    async fn create_table(
```

Comment:
We should include metadata location for most endpoints, including the old metadata location if it changed. We already have this for the `commit_transaction` endpoint, but not for views.